transport/config/config.go

```diff
@@ -0,0 +1,146 @@
+	// Default 100
+	MaxIdleConnsPerHost int `yaml:"MaxIdleConnsPerHost" validate:"min=0"`
+	// IdleConnTimeout see: https://golang.org/pkg/net/http/#Transport
+	// Default 0 (no limit)
+	IdleConnTimeout metrics.Interval `yaml:"IdleConnTimeout"`
+	// ResponseHeaderTimeout see: https://golang.org/pkg/net/http/#Transport
+	// Default 5s (no limit)
+	ResponseHeaderTimeout metrics.Interval `yaml:"ResponseHeaderTimeout"`
+	// DisableKeepAlives see: https://golang.org/pkg/net/http/#Transport
+	// Default false
+	DisableKeepAlives bool `yaml:"DisableKeepAlives"`
+}
+
+// ClientTransportMatchers properties
+type ClientTransportMatchers struct {
+	Method     string `yaml:"Method" validate:"max=64"`
+	Path       string `yaml:"Path" validate:"max=64"`
+	QueryParam string `yaml:"QueryParam" validate:"max=64"`
+}
+
+// MatchersCompiledRules properties
+type MatchersCompiledRules struct {
+	MethodRegexp     *regexp.Regexp
+	PathRegexp       *regexp.Regexp
+	QueryParamRegexp *regexp.Regexp
+	IsCompiled       bool
+}
+
+// Transport properties
+type Transport struct {
+	Name                  string                  `yaml:"Name"`
+	Matchers              ClientTransportMatchers `yaml:"Matchers"`
+	MatchersCompiledRules MatchersCompiledRules
+	Details               ClientTransportDetail `yaml:"Details"`
+}
+
+// Transports map with Transport
+type Transports []Transport
+
+// compileRule
+func (t *Transport) compileRule(regexpRule string) (compiledRule *regexp.Regexp, err error) {
+	if len(regexpRule) > 0 {
+		compiledRule, err = regexp.Compile(regexpRule)
+	}
+	return
+}
+
+// transportFlags for internal matching func
+type transportFlags struct {
+	declared bool
+	matched  bool
+	empty    bool
+}
+
+// compileRules prepares precompiled regular expressions for rules
+func (t *Transport) compileRules() error {
+	if !t.MatchersCompiledRules.IsCompiled {
+		if len(t.Matchers.Method) > 0 {
+			var err error
+			t.MatchersCompiledRules.MethodRegexp, err = t.compileRule(t.Matchers.Method)
+			if err != nil {
+				return fmt.Errorf("compileRule for Client->Transport->Trigger->Method error: %q", err)
+			}
+		}
+		if len(t.Matchers.Path) > 0 {
+			var err error
+			t.MatchersCompiledRules.PathRegexp, err = t.compileRule(t.Matchers.Path)
+			if err != nil {
+				return fmt.Errorf("compileRule for Client->Transport->Trigger->Path error: %q", err)
+			}
+		}
+		if len(t.Matchers.QueryParam) > 0 {
+			var err error
+			t.MatchersCompiledRules.QueryParamRegexp, err = t.compileRule(t.Matchers.QueryParam)
+			if err != nil {
+				return fmt.Errorf("compileRule for Client->Transport->Trigger->QueryParam error: %q", err)
+			}
+		}
+		t.MatchersCompiledRules.IsCompiled = true
+	}
+	return nil
+}
+
+// GetMatchedTransport returns first details matching with rules from Matchers by arguments: method, path, queryParam
+func (t *Transports) GetMatchedTransport(method, path, queryParam string) (matchedTransport Transport, matchedTransportName string, ok bool) {
```

Comment:
None of callers use first returned value, maybe it should not be returrned at all?